Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java

```diff
@@ -104,59 +104,114 @@ public WebApp createWebAppOnWindows(@NotNull WebAppSettingModel model) throws Ex
-                    .withRegion(model.getRegion())
-                    .withNewResourceGroup(model.getResourceGroup())
-                    .withPricingTier(pricing)
-                    .withOperatingSystem(OperatingSystem.WINDOWS);
-            withCreateWebApp = azure.webApps().define(model.getWebAppName())
-                    .withRegion(model.getRegion())
-                    .withNewResourceGroup(model.getResourceGroup())
-                    .withNewWindowsPlan(withCreatePlan);
+        PricingTier pricingTier = new PricingTier(tierSize[0], tierSize[1]);
+
+        boolean isCreatingNewResourceGroup = model.isCreatingResGrp();
+        AppServicePlan.DefinitionStages.WithGroup withGroup = azure
+            .appServices()
+            .appServicePlans()
+            .define(model.getAppServicePlanName())
+            .withRegion(model.getRegion());
+
+        AppServicePlan.DefinitionStages.WithPricingTier withPricingTier;
+        String resourceGroup = model.getResourceGroup();
+        if (isCreatingNewResourceGroup) {
+            withPricingTier = withGroup.withNewResourceGroup(resourceGroup);
         } else {
-            withCreatePlan = azure.appServices().appServicePlans()
-                    .define(model.getAppServicePlanName())
-                    .withRegion(model.getRegion())
-                    .withExistingResourceGroup(model.getResourceGroup())
-                    .withPricingTier(pricing)
-                    .withOperatingSystem(OperatingSystem.WINDOWS);
-            withCreateWebApp = azure.webApps().define(model.getWebAppName())
-                    .withRegion(model.getRegion())
-                    .withExistingResourceGroup(model.getResourceGroup())
-                    .withNewWindowsPlan(withCreatePlan);
+            withPricingTier = withGroup.withExistingResourceGroup(resourceGroup);
         }
-        return withCreateWebApp;
+
+        return withPricingTier.withPricingTier(pricingTier);
+    }
+
+    private WebApp.DefinitionStages.WithCreate withCreateNewWindowsServicePlan(
+            @NotNull Azure azure, @NotNull WebAppSettingModel model) throws Exception {
+
+        boolean isCreatingNewResGrp = model.isCreatingResGrp();
+        String resourceGroup = model.getResourceGroup();
+
+        AppServicePlan.DefinitionStages.WithCreate withCreatePlan = prepareWithOperatingSystem(azure, model)
+            .withOperatingSystem(OperatingSystem.WINDOWS);
+        WebApp.DefinitionStages.NewAppServicePlanWithGroup appWithGroup = azure
+            .webApps()
+            .define(model.getWebAppName())
+            .withRegion(model.getRegion());
+
+        if (isCreatingNewResGrp) {
+            return appWithGroup.withNewResourceGroup(resourceGroup).withNewWindowsPlan(withCreatePlan);
+        }
+        return appWithGroup.withExistingResourceGroup(resourceGroup).withNewWindowsPlan(withCreatePlan);
+    }
+
+    private WebApp.DefinitionStages.WithDockerContainerImage withCreateNewLinuxServicePlan(
```

Comment:
Duplicated codes compared with withCreateNewWindowsServicePlan().

Reply:
Use helper methods to handle the duplicated codes.